client/java/src/main/java/io/openlineage/client/metrics/StatsDMetricsBuilder.java

```diff
@@ -0,0 +1,27 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.client.metrics;
+
+import io.micrometer.statsd.StatsdMeterRegistry;
```

Comment:
I think it will work the same way as Spark visitor hasClasses methods: have them as compileOnly dependencies, and make static check for their existence, why would it be different?